handout/a11y/key-concerns/semantic-markup.md

```diff
@@ -0,0 +1,158 @@
+  message = 'Alert!';
+
+  @Input()
+  label = 'Notify';
+
+  constructor(private notification: NotificationService) { }
+
+  @HostListener('click', [])
+  notify() {
+    this.notification.notify(this.message)
+  }
+}
+```
+[View Example](https://plnkr.co/edit/aAjNnmeaEPbdfIWo9hPT?p=preview)
+
+This lets you create a component that has the same semantics as a `button` element to screen readers and browsers, but now have the opportunity to fully control the styling of that component as well as inject your own custom logic.
+
+
+### ARIA attributes
+
+Some native html tags have attributes that providers extra context on what's being displayed on the browser: The `img` tag's `alt` attribute for example, lets the reader know what is being shown using a short description. However, native tags don't cover all cases. This is where ARIA fits in. ARIA attributes can provide context on what roles specific elements have in the application or can provide context on how some elements are related to other elements on the document. One example of this is modals. Native modals provided by different platforms such as web browsers, often have limited customization options which can make for a poor experience in some cases. This necessitates the creation of custom modals. A modal component can be given the `role` of [dialog](https://www.w3.org/TR/wai-aria/roles#dialog) or [alertdialog](https://www.w3.org/TR/wai-aria/roles#alertdialog) to let the browser know that that component is acting as a modal.  Within the component we can use the ARIA attributes `aria-labelledby` and `aria-described` by to let readers know which elements within those modals say what the title of that modal is and a longer description of what exactly the user is being shown within that modal.
+
+*app.component.ts*
+```
+@Component({
+	selector: 'ngc2-app',
+	template: `
+	  <ngc2-notification-button
+	    message="Hello!"
+	    label="Greeting"
+	    role="button">
+	  </ngc2-notification-button>
+	  <ngc2-modal
+	    [title]="modal.title"
+	    [description]="modal.description"
+	    [visible]="modal.visible"
+	    (close)="modal.close()">
+	  </ngc2-modal>
+	`
+})
+export class AppComponent {
+  constructor(private modal: ModalService) { }
+}
+```
+
+*notification-button.component.ts*
+```
+@Component({
+  selector: 'ngc2-modal',
+  template: `
+    <div
+      role="dialog"
+      aria-labelledby="modal-title"
+      aria-describedby="modal-description">
+      <div id="modal-title">{{title}}</div>
+      <p id="modal-description">{{description}}</p>
+      <button (click)="close.emit()">OK</button>
+    </div>
+  `
+})
+export class ModalComponent {
+  ...
+}
+```
+[View Example](https://plnkr.co/edit/Vvu62nDZ18IkqiAop2A9?p=preview)
+
+
+## Additional Information
+This information covered highlights some of the key concepts and considerations regarding semantic markup and ARIA details. More information is available in the [WAI-ARIA specification](https://www.w3.org/TR/wai-aria/).
```

Comment:
Possibly worth adding somewhere that ARIA shouldn't be relied on to provide all accessibility functionality, and the bulk of an application's accessibility doesn't come from ARIA tagging.